Source/GTMSessionFetcherService.m

```diff
@@ -374,17 +374,19 @@ - (void)startFetcher:(GTMSessionFetcher *)fetcher {
 }
 
 // Internal utility. Returns a fetcher's delegate if it's a dispatcher, or nil if the fetcher
-// is its own delegate and has no dispatcher.
+// is its either own delegate and has no dispatcher or swizzled alien.
 - (GTMSessionFetcherSessionDelegateDispatcher *)delegateDispatcherForFetcher:(GTMSessionFetcher *)fetcher {
   GTMSessionCheckNotSynchronized(self);
 
   NSURLSession *fetcherSession = fetcher.session;
   if (fetcherSession) {
     id<NSURLSessionDelegate> fetcherDelegate = fetcherSession.delegate;
-    BOOL hasDispatcher = (fetcherDelegate != nil && fetcherDelegate != fetcher);
-    if (hasDispatcher) {
+      BOOL maybeHasDispatcher = (fetcherDelegate != nil && fetcherDelegate != fetcher);
+      
       GTMSESSION_ASSERT_DEBUG([fetcherDelegate isKindOfClass:[GTMSessionFetcherSessionDelegateDispatcher class]],
```

Comment:
Why do the assert but let the code work when that isn't the case?  What are you trying to allow with the new test that you still want the assert also?

Reply:
True. I've kept this assert to bring focus to this issue. With the new code this assert should be replaced with the warning. Changing that.